src/ripple/protocol/SOTemplate.h

```diff
@@ -61,50 +75,44 @@ class SOElement
 class SOTemplate
 {
 public:
-    using list_type = std::vector <std::unique_ptr <SOElement const>>;
-    using iterator_range = boost::iterator_range<list_type::const_iterator>;
+    SOTemplate(SOTemplate&& other) = default;
 
-    /** Create an empty template.
-        After creating the template, call @ref push_back with the
-        desired fields.
-        @see push_back
+    /** Create a template populated with all fields.
+        After creating the template fields cannot be
+        added, modified, or removed.
     */
-    SOTemplate () = default;
+    SOTemplate (std::initializer_list<SOElement> uniqueFields,
+        std::initializer_list<SOElement> commonFields = {});
 
-    SOTemplate(SOTemplate&& other)
-        : mTypes(std::move(other.mTypes))
-        , mIndex(std::move(other.mIndex))
+    /* Provide for the enumeration of fields */
+    std::vector<SOElement>::const_iterator begin() const
```

Comment:
I use `begin/end` myself, but lately we've been adding `cbegin/cend` to classes with `begin/end` as well. May as well do it here to be consistent.